### ISSUE TITLE
Encoder Use (EMOD) in MBS file

### ISSUE THREAD

**Issue by Myzhar** [g/roboteq#13]
Hello,

I'm not sure about the correctness of the following line:
**https://github.com/g/roboteq/blob/indigo-devel/roboteq_driver/mbs/script.mbs#L30**

I don't understand why you set EMOD to 50, while according to manual (p210) it should be
aa + mot1 \* 16 + mot2 \* 32

so 

18 for motor 1 with feedback
34 for motor 2 with feedback

Walter

**Comment by Myzhar** [comment]
Hi,

I have found a post on the RoboteQ's forum that confirms my hypothesis:
http://www.roboteq.com/index.php/forum/9-controller-configuration-operation/29529115-encoder-setup-after-controller-reset

**Comment by romainreignier** [comment]
I would be also interested in the answer.

**Comment by rbonghi** [comment]
I @romainreignier I've fixed the @Myzhar with a new node in ROS https://github.com/rbonghi/roboteq_control This node works with ROS control and with the dynamic reconfigurator you can dinamically update the roboteq configuration. It has look like RoboRun+

**Comment by romainreignier** [comment]
@rbonghi yes, I have seen your node. The concept to use only the serial comm and no script and the dynamic reconfigure for everything is very interesting.
But I do not like in your code that you call the `nh.getParam()` every time in each callback because it takes some time so it is not very efficient. And I do not have the time right now to modify it.

**Comment by rbonghi** [comment]
yes, I know only this way :-( to dynamically update the the parameters when the motors run. If you know other solution I am interested to change this approach.

At this time I've tested with a Jetson TX2 and works better without delay.

**Comment by romainreignier** [comment]
This issue is not the right place for this discussion. But usually we use class members updated in the dynamic_reconfigure callback for this.
